sdk/src/main/java/com/silanis/esl/sdk/builder/ProviderBuilder.java

```diff
@@ -0,0 +1,47 @@
+package com.silanis.esl.sdk.builder;
+
+import com.silanis.esl.sdk.Provider;
+
+import java.util.Map;
+
+public class ProviderBuilder {
+    private String provides;
+    private Map<String, Object> data;
+    private String id;
+    private String name;
+
+    private ProviderBuilder() {}
+
+    public static ProviderBuilder newProvider() {
+        return new ProviderBuilder();
+    }
+    
+    public ProviderBuilder withId(String value ) {
+        id = value;
+        return this;
+    }
```

Comment:
Please don't name the parameter as `value`, but name it as the actual parameter name as below.

```
    public ProviderBuilder withId( String id ) {
        this.id = id;
        return this;
    }
```

Reply:
Got it, will be updated.